go/mysql/server.go

```diff
@@ -294,10 +315,37 @@ func (l *Listener) handle(conn net.Conn, connectionID uint32, acceptTime time.Ti
 		log.Warningf("Slow connection from %s: %v", c, connectTime)
 	}
 
+	packetCh := make(chan []byte, 1)
+	errorCh := make(chan error)
+	mu := sync.Mutex{}
+	var data []byte
+
+	go func() {
+		for {
+			mu.Lock()
+			c.sequence = 0
+			data, err := c.readEphemeralPacket()
+
+			if err != nil {
+				errorCh <- err
+				return
+			}
+
+			packetCh <- data
+		}
+	}()
```

Comment:
If it's not obvious, the point of moving `readEphemeralPacket()` into its own goroutine is so that `shutdownCh` can preempt it (since it blocks until a packet has been read or the connection has closed). The `packetCh` and `errorCh` channels feel a little gross to me, but I'm not sure if theres a better way (still somewhat of a Go newbie). I can add a comment explaining all this if it helps.